diff -rupN clean/src/wifi/model/wifi-mac.cc mod/src/wifi/model/wifi-mac.cc
--- clean/src/wifi/model/wifi-mac.cc	2013-11-15 22:50:32.000000000 +0100
+++ mod/src/wifi/model/wifi-mac.cc	2015-04-09 15:30:59.736125286 +0200
@@ -331,13 +331,13 @@ WifiMac::Configure80211b (void)
 void
 WifiMac::Configure80211g (void)
 {
-  SetSifs (MicroSeconds (10));
+  //SetSifs (MicroSeconds (10));
   // Note no support for Short Slot Time as yet
-  SetSlot (MicroSeconds (20));
-  SetEifsNoDifs (MicroSeconds (10 + 304));
-  SetPifs (MicroSeconds (10 + 20));
-  SetCtsTimeout (MicroSeconds (10 + 304 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
-  SetAckTimeout (MicroSeconds (10 + 304 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
+  //SetSlot (MicroSeconds (20));
+  //SetEifsNoDifs (MicroSeconds (10 + 304));
+  //SetPifs (MicroSeconds (10 + 20));
+  //SetCtsTimeout (MicroSeconds (10 + 304 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
+  //SetAckTimeout (MicroSeconds (10 + 304 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
 }
 
 void
@@ -367,8 +367,8 @@ WifiMac::Configure80211n_2_4Ghz (void)
 {
   Configure80211g ();
   SetRifs(MicroSeconds (2));
-  SetCtsTimeout (MicroSeconds (10 + 52 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
-  SetAckTimeout (MicroSeconds (10 + 52 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
+  //SetCtsTimeout (MicroSeconds (10 + 52 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
+  //SetAckTimeout (MicroSeconds (10 + 52 + 20 + GetDefaultMaxPropagationDelay ().GetMicroSeconds () * 2));
 }
 void
 WifiMac::Configure80211n_5Ghz (void)
diff -rupN clean/src/wifi/model/wifi-phy.cc mod/src/wifi/model/wifi-phy.cc
--- clean/src/wifi/model/wifi-phy.cc	2013-11-15 22:50:32.000000000 +0100
+++ mod/src/wifi/model/wifi-phy.cc	2015-03-23 16:54:40.000000000 +0100
@@ -884,12 +884,18 @@ WifiMode
 WifiPhy::GetOfdmRate1_5MbpsBW5MHz ()
 {
   static WifiMode mode =
-    WifiModeFactory::CreateWifiMode ("OfdmRate1_5MbpsBW5MHz",
+   /* WifiModeFactory::CreateWifiMode ("OfdmRate1_5MbpsBW5MHz",
                                      WIFI_MOD_CLASS_OFDM,
                                      true,
                                      5000000, 1500000,
                                      WIFI_CODE_RATE_1_2,
-                                     2);
+                                     2);*/
+      WifiModeFactory::CreateWifiMode ("OfdmRate1_5MbpsBW5MHz",
+    		  	  	  	  	  	  	  WIFI_MOD_CLASS_HT,
+		                             true,
+		                             20000000, 78500000,
+		                             WIFI_CODE_RATE_3_4,
+		                             16);
   return mode;
 }
 
@@ -897,12 +903,18 @@ WifiMode
 WifiPhy::GetOfdmRate2_25MbpsBW5MHz ()
 {
   static WifiMode mode =
-    WifiModeFactory::CreateWifiMode ("OfdmRate2_25MbpsBW5MHz",
+    /*WifiModeFactory::CreateWifiMode ("OfdmRate2_25MbpsBW5MHz",
                                      WIFI_MOD_CLASS_OFDM,
                                      false,
                                      5000000, 2250000,
                                      WIFI_CODE_RATE_3_4,
-                                     2);
+                                     2);*/
+	WifiModeFactory::CreateWifiMode ("OfdmRate2_25MbpsBW5MHz",
+									WIFI_MOD_CLASS_HT,
+		  		                    true,
+		  		                    20000000, 104000000,
+		  		                    WIFI_CODE_RATE_2_3,
+		  		                    64);
   return mode;
 }
 
@@ -910,12 +922,18 @@ WifiMode
 WifiPhy::GetOfdmRate3MbpsBW5MHz ()
 {
   static WifiMode mode =
-    WifiModeFactory::CreateWifiMode ("OfdmRate3MbpsBW5MHz",
+    /*WifiModeFactory::CreateWifiMode ("OfdmRate3MbpsBW5MHz",
                                      WIFI_MOD_CLASS_OFDM,
                                      true,
                                      5000000, 3000000,
                                      WIFI_CODE_RATE_1_2,
-                                     4);
+                                     4);*/
+		  WifiModeFactory::CreateWifiMode ("OfdmRate3MbpsBW5MHz",
+				  	  	  	  	  	  	  WIFI_MOD_CLASS_HT,
+		  		  		                    true,
+		  		  		                    20000000, 117000000,
+		  		  		                    WIFI_CODE_RATE_3_4,
+		  		  		                    64);
   return mode;
 }
 
@@ -923,12 +941,18 @@ WifiMode
 WifiPhy::GetOfdmRate4_5MbpsBW5MHz ()
 {
   static WifiMode mode =
-    WifiModeFactory::CreateWifiMode ("OfdmRate4_5MbpsBW5MHz",
+   /* WifiModeFactory::CreateWifiMode ("OfdmRate4_5MbpsBW5MHz",
                                      WIFI_MOD_CLASS_OFDM,
                                      false,
                                      5000000, 4500000,
                                      WIFI_CODE_RATE_3_4,
-                                     4);
+                                     4);*/
+		  WifiModeFactory::CreateWifiMode ("OfdmRate4_5MbpsBW5MHz",
+				  	  	  	  	  	  	  WIFI_MOD_CLASS_HT,
+		  		  		                    true,
+		  		  		                    20000000, 130000000,
+		  		  		                    WIFI_CODE_RATE_5_6,
+		  		  		                    64);
   return mode;
 }
 
@@ -1163,7 +1187,7 @@ WifiPhy::GetOfdmRate65MbpsBW20MHzShGi ()
                                      WIFI_MOD_CLASS_HT,
                                      false,
                                      20000000, 65000000,
-                                     WIFI_CODE_RATE_3_4,
+                                     WIFI_CODE_RATE_5_6,
                                      64);
   return mode;
 }
